standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java

```diff
@@ -5811,11 +5811,11 @@ public void performTimeOuts() {
         s = sqlGenerator.addLimitClause(10 * TIMED_OUT_TXN_ABORT_BATCH_SIZE, s);
 
         LOG.debug("Going to execute query <{}>", s);
-        List<List<Long>> timedOutTxns = jdbcResource.getJdbcTemplate().query(s, rs -> {
-          List<List<Long>> txnbatch = new ArrayList<>();
-          List<Long> currentBatch = new ArrayList<>(TIMED_OUT_TXN_ABORT_BATCH_SIZE);
+        List<List<Pair<Long, Integer>>> timedOutTxns = jdbcResource.getJdbcTemplate().query(s, rs -> {
+          List<List<Pair<Long, Integer>>> txnbatch = new ArrayList<>();
+          List<Pair<Long, Integer>> currentBatch = new ArrayList<>(TIMED_OUT_TXN_ABORT_BATCH_SIZE);
           while (rs.next()) {
-            currentBatch.add(rs.getLong(1));
+            currentBatch.add(Pair.of(rs.getLong(1),rs.getInt(2)));
```

Comment:
sorry, looks like Map<Long, TxnType> was a valid choice, so you can easily access jus the ids
````
currentBatch.put(Pair.of(rs.getLong(1), TxnType.findByValue(rs.getInt(2))); 
````

Reply:
Yes, reverted to Map